x-pack/plugins/security_solution/public/resolver/store/data/visible_entities.test.ts

```diff
@@ -0,0 +1,159 @@
+        event_subtype_full: 'creation_event',
+        unique_pid: 2,
+        unique_ppid: 1,
+      },
+    });
+    processD = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 3,
+        unique_ppid: 2,
+      },
+    });
+    processE = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 4,
+        unique_ppid: 3,
+      },
+    });
+    processF = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 5,
+        unique_ppid: 4,
+      },
+    });
+    processF = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'creation_event',
+        unique_pid: 6,
+        unique_ppid: 5,
+      },
+    });
+    processG = mockProcessEvent({
+      endgame: {
+        event_type_full: 'process_event',
+        event_subtype_full: 'termination_event',
+        unique_pid: 7,
+        unique_ppid: 6,
+      },
+    });
+    store = createStore(resolverReducer, undefined);
+  });
+  describe('when rendering a large tree with a small viewport', () => {
+    beforeEach(() => {
+      const events: ResolverEvent[] = [processA, processB];
+      const action: ResolverAction = {
+        type: 'serverReturnedResolverData',
+        events,
+        stats: new Map(),
+      };
+      const cameraAction: ResolverAction = { type: 'userSetRasterSize', payload: [300, 200] };
+      store.dispatch(action);
+      store.dispatch(cameraAction);
+    });
+    it('the visibleProcessNodePositions list should only include 2 nodes', () => {
+      const { processNodePositions } = visibleProcessNodePositionsAndEdgeLineSegments(
+        store.getState()
+      )(Date.now());
+      expect(processNodePositions.length).toEqual(2);
+    });
+    it('the visibleEdgeLineSegments list should only include 1 lines', () => {
+      const { connectingEdgeLineSegments } = visibleProcessNodePositionsAndEdgeLineSegments(
+        store.getState()
+      )(Date.now());
+      expect(connectingEdgeLineSegments.length).toEqual(1);
+    });
+  });
+  describe('when rendering a large tree with a large viewport', () => {
+    beforeEach(() => {
+      const events: ResolverEvent[] = [
+        processA,
+        processB,
+        processC,
+        processD,
+        processE,
+        processF,
+        processG,
+      ];
+      const action: ResolverAction = {
+        type: 'serverReturnedResolverData',
+        events,
+        stats: new Map(),
+      };
+      const cameraAction: ResolverAction = { type: 'userSetRasterSize', payload: [1000, 1000] };
+      store.dispatch(action);
+      store.dispatch(cameraAction);
+    });
+    it('the visibleProcessNodePositions list should include all lines', () => {
```

Comment:
:grey_question: I think you intended `list should include all lines` to be `list should include all _processNodePositions`